Fig.Core/InvariantStringConverter.cs

```diff
@@ -13,7 +13,16 @@ public T Convert<T>(string value)
         public object Convert(string value, Type targetType)
         {
             var typeConverter = TypeDescriptor.GetConverter(targetType);
-            return typeConverter.ConvertFromInvariantString(value);
+
+            var convertedValue = typeConverter.ConvertFromInvariantString(value);
+
+            // If target type is an enum, check that the parsed string value is a valid value
+            if (targetType.IsEnum && !Enum.IsDefined(targetType, convertedValue))
+            {
+                throw new ArgumentException($"The string '{nameof(value)}' provided is not a valid Enum value.");
```

Comment:
nameof(value) here will literally yield the string "value".  Probably not what you intended?

Reply:
It's what I intended but I can aslo see why that might be confusing.  Will reword to make it little more clear.

Reply:
No worries, I would have included the actual value and also provided a list of valid options. It's useful to have the actual data in the output but obviously could impose a huge security risk. But in any case, there's an ArgumentException constructor overload that takes a message and the argument name.